source/module_esolver/esolver_fp.cpp

```diff
@@ -252,6 +254,31 @@ void ESolver_FP::after_scf(const int istep)
                 this->pelec->pot->get_fixed_v());
         }
     }
+
+    // 5) write ELF
+    if (PARAM.inp.out_elf)
+    {
+        this->pelec->charge->cal_elf = true;
+        Symmetry_rho srho;
+        for (int is = 0; is < GlobalV::NSPIN; is++)
+        {
+            srho.begin(is, *(this->pelec->charge), this->pw_rho, GlobalC::ucell.symm);
+        }
```

Comment:
You should be careful that pw_rhod is the grids for charge/potential, while pw_rho is the smooth grid in uspp.

Reply:
Thanks, I have updated it.